src/domains/messages-app/12_function_service_messages.tf

```diff
@@ -0,0 +1,299 @@
+    "Microsoft.Storage",
+  ]
+
+  delegation = {
+    name = "default"
+    service_delegation = {
+      name    = "Microsoft.Web/serverFarms"
+      actions = ["Microsoft.Network/virtualNetworks/subnets/action"]
+    }
+  }
+}
+
+module "function_service_messages" {
+  count  = var.function_service_messages_enabled ? 1 : 0
+  source = "github.com/pagopa/terraform-azurerm-v3//function_app?ref=v7.70.1"
+
+  resource_group_name = azurerm_resource_group.service_messages_rg.name
+  name                = format("%s-service-messages-fn", local.product)
+  location            = azurerm_resource_group.service_messages_rg.location
+  health_check_path   = "/api/v1/info"
+  domain              = "MESSAGES"
+
+  node_version                             = "18"
+  runtime_version                          = "~4"
+  always_on                                = true
+  application_insights_instrumentation_key = data.azurerm_application_insights.application_insights.instrumentation_key
+
+  app_service_plan_info = {
+    kind                         = "Linux"
+    sku_tier                     = "PremiumV3"
+    sku_size                     = "P1v3"
+    maximum_elastic_worker_count = 0
+    worker_count                 = null
+    zone_balancing_enabled       = false
+  }
+
+  storage_account_info = {
+    account_kind                      = "StorageV2"
+    account_tier                      = "Standard"
+    account_replication_type          = "GZRS"
+    access_tier                       = "Hot"
+    advanced_threat_protection_enable = true
+    use_legacy_defender_version       = true
+  }
+
+  app_settings = merge(
+    local.function_service_messages.app_settings,
+  )
+
+  subnet_id = module.function_service_messages_snet.id
+
+  allowed_subnets = [
+    module.function_service_messages_snet.id,
+    data.azurerm_subnet.apim_snet.id,
+  ]
+
+  allowed_ips = concat(
+    [],
+    local.app_insights_ips_west_europe,
+  )
+
+  action = [
+    {
+      action_group_id    = data.azurerm_monitor_action_group.error_action_group.id
+      webhook_properties = null
+    }
+  ]
+
+  client_certificate_mode = "Required"
+
+  tags = var.tags
+}
+
+module "function_service_messages_staging_slot" {
+  count  = var.function_service_messages_enabled ? 1 : 0
+  source = "github.com/pagopa/terraform-azurerm-v3//function_app_slot?ref=v7.70.1"
+
+  name                = "staging"
+  location            = var.location
+  resource_group_name = azurerm_resource_group.service_messages_rg.name
+  function_app_id     = module.function_service_messages.id
+  app_service_plan_id = module.function_service_messages.app_service_plan_id
+  health_check_path   = "/api/v1/info"
+
+  node_version                             = "18"
+  runtime_version                          = "~4"
+  always_on                                = true
+  application_insights_instrumentation_key = data.azurerm_application_insights.application_insights.instrumentation_key
+
+  app_settings = merge(
```

Comment:
do we have to disable some functions in the staging slot? ie. `CosmosRemoteContentMessageConfigurationChangeFeed` ?

Reply:
Nope. Not here. That's going to be migrated under function-cqrs. 
By the way, I was just making a PR to disable it on the other staging functions.